src/absil/ilreflect.fs

```diff
@@ -336,11 +336,11 @@ let convTypeRefAux (cenv:cenv) (tref:ILTypeRef) =
             | None ->
                 let asmName    = convAssemblyRef asmref
                 FileSystem.AssemblyLoad(asmName)
-        let typT       = assembly.GetType(qualifiedName)
+        let typT = assembly.GetType(qualifiedName, throwOnError=true)
         typT |> nonNull "convTypeRefAux" 
     | ILScopeRef.Module _ 
     | ILScopeRef.Local _ ->
-        let typT = Type.GetType(qualifiedName,true) 
+        let typT = Type.GetType(qualifiedName, throwOnError=true) 
         typT |> nonNull "convTypeRefAux" 
```

Comment:
It might not be needed anymore. Here is the confusing bit from the link you cite:

> GetType only works on assemblies loaded from disk. If you call GetType to look up a type defined in a dynamic assembly defined using the System.Reflection.Emit services, you might get inconsistent behavior. The behavior depends on whether the dynamic assembly is persistent, that is, created using the RunAndSave or Save access modes of the System.Reflection.Emit.AssemblyBuilderAccess enumeration. If the dynamic assembly is persistent and has been written to disk before GetType is called, the loader finds the saved assembly on disk, loads that assembly, and retrieves the type from that assembly. If the assembly has not been saved to disk when GetType is called, the method returns null. GetType does not understand transient dynamic assemblies; therefore, calling GetType to retrieve a type in a transient dynamic assembly returns null.

Reading the source, I'm not entirely sure the method never returns `null`.

http://referencesource.microsoft.com/#mscorlib/system/reflection/assembly.cs,1454

So to be on the safe side, I still left the null check there.